libraries/rush-lib/src/api/CustomTipsConfiguration.ts

```diff
@@ -56,11 +46,66 @@ export interface ICustomTipItemJson {
  *
  * @beta
  */
-export type CustomTipId =
-  // The rush.json "ensureConsistentVersions" validation.
-  | 'TIP_RUSH_INCONSISTENT_VERSIONS'
-  // In the future, plugins can contribute other strings.
-  | string;
+export enum CustomTipIdEnum {
+  TIP_RUSH_INCONSISTENT_VERSIONS = 'TIP_RUSH_INCONSISTENT_VERSIONS',
+  TIP_PNPM_NO_MATCHING_VERSION = 'TIP_PNPM_NO_MATCHING_VERSION'
+}
+
+/**
+ * The severity of a custom tip.
+ * It determines the printing severity ("Error" = red, "Warning" = yellow, "Info" = normal).
+ */
+enum CustomTipSeverity {
+  Warning = 'Warning',
+  Error = 'Error',
+  Info = 'Info'
+}
+
+/**
+ * The type of the custom tip.
+ *
+ * @remarks
+ * There might be types like `git` in the future.
+ */
+export enum CustomTipType {
+  rush = 'rush',
+  pnpm = 'pnpm'
+}
+
+/**
+ * Metadata for a custom tip.
+ *
+ * @remarks
+ * This differs from the  {@link ICustomTipItemJson} interface in that these are not configurable by the user; it's the inherent state of a custom tip.
+ * For example, the custom tip for `ERR_PNPM_NO_MATCHING_VERSION` has a inherent severity of `Error`, and a inherent match function that rush maintainer defines.
+ * @beta
+ *
+ */
+export interface ICustomTipInfo {
+  tipId: CustomTipIdEnum;
+  /**
+   * (Required) The severity of the custom tip. It will determine the printing severity ("Error" = red, "Warning" = yellow, "Info" = normal).
```

Comment:
That this is required is implied by the type.

```suggestion
   * The severity of the custom tip. It will determine the printing severity ("Error" = red, "Warning" = yellow, "Info" = normal).
```